log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java

```diff
@@ -623,7 +623,12 @@ public Configuration setConfiguration(final Configuration config) {
                 map.putIfAbsent("hostName", "unknown");
             }
             map.putIfAbsent("contextName", contextName);
-            config.start();
+            try {
+            	config.start();
+            } catch (Exception e) {
+            	config.stop();
+            	return configuration;
+            }
```

Comment:
We are swallowing the exception here. Shouldn't we check if it matches first and otherwise raise it?

Reply:
I will change this part and raise the exception always